canlink/writer/writer.go

```diff
@@ -0,0 +1,68 @@
+package writer
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// Responsible for managing the trace file
+type Writer struct {
+	traceFile       *os.File
+	fileExtention   string
+	convertToString func(*zap.Logger, *TimestampedFrame) string
+	l               *zap.Logger
+}
+
+func NewWriter(l *zap.Logger, fileExtention string) *Writer {
+	convertToStringMapping := map[string]func(*zap.Logger, *TimestampedFrame) string{
```

Comment:
this map prevents developers from defining their own converters since the only supported ones are json and ascii. (i'm thinking about how we want to make this available for others to use).

You could define an interface representing the `Ascii` and `Json` functions which include the `to_string` method and a `.get_extension() -> str` function. Then, you would pass one of those interfaces to `NewWriter`. This would make it possible for someone else to define a `CSV` converter and just pass it into NewWriter